cli/apiserver/provider.go

```diff
@@ -0,0 +1,167 @@
+package apiserver
+
+import (
+	"archive/zip"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/up9inc/mizu/cli/logger"
+	"github.com/up9inc/mizu/cli/uiUtils"
+	"github.com/up9inc/mizu/shared"
+	"io/ioutil"
+	core "k8s.io/api/core/v1"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+type apiServerProvider struct {
+	url     string
+	isReady bool
+}
+
+var Provider = apiServerProvider{}
+
+func (provider *apiServerProvider) Init(url string, retries int) error {
```

Comment:
I would name it InitWithRetries.
because when I read it from outside (when we call this function) it is not clear what is the number after the url. 
it is not mandatory but maybe easier if we see the word retries in the function name